### PR TITLE
#2633 jacksonxml: add test coverage

### PR DESCRIPTION
**Pull request by ffang**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by ppalaga — COMMENTED** [review]
Thanks, the test coverage is highly appreciated. Except for the two minor comments inline, could you please move the Jackson XML related stuff to separate test classes (JacksonXmlTest and JacksonXmlIT) and to a separate package (e.g. `org.apache.camel.quarkus.component.dataformats.jackson.xml`) in the test app? That would make it easier for the future readers to understand which parts are testing which extensions.

File: integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/json/JsonDataformatsRoute.java [hunk 1]
```diff
@@ -88,6 +96,97 @@ public boolean shouldSkipClass(Class<?> clazz) {
                 .unmarshal()
                 .jacksonxml(PojoA.class);
 
+        JacksonXMLDataFormat unmarshalTypeHeaderFormat = new JacksonXMLDataFormat();
+        unmarshalTypeHeaderFormat.setAllowUnmarshallType(true);
+        from("direct:jacksonxml-unmarshal-type-header").unmarshal(unmarshalTypeHeaderFormat);
+
+        JacksonDataFormat listFormat = new JacksonDataFormat(TestPojo.class);
```
Comment (aldettinger) [inline integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/json/JsonDataformatsRoute.java:103]: Shouldn't this one be `JacksonXMLDataFormat` too ?

**Comment by aldettinger** [comment]
Indeed, splitting more between jackson and jacksonxml would make the code more readable.

Concerning jackson features, I was first in the idea that it is the responsibility of quarkus-jackson to test them. But given, we already found an issue with @JsonView, I wonder if it would make sense to test @JsonInclude also ?

File: integration-tests/dataformats-json/src/main/resources/routes/jackson-routes.xml [hunk 2]
```diff
@@ -40,4 +40,57 @@
...
+    <route>
+      <from uri="direct:jacksonxml-marshal-spring-enablefeature"/>
+      <marshal>
+        <jacksonxml unmarshalTypeName="org.apache.camel.quarkus.component.dataformats.json.model.TestPojo"
+            enableFeatures="WRAP_ROOT_VALUE"/>
+      </marshal>
+    </route>
+
+    <route>
+      <from uri="direct:spring-in"/>
```
[the remark this answers is not in the document]
Comment (ffang) [reply integration-tests/dataformats-json/src/main/resources/routes/jackson-routes.xml:58]: sure, will revise accordingly.

**Review by ffang — COMMENTED** [review]
> Indeed, splitting more between jackson and jacksonxml would make the code more readable.
> 
> Concerning jackson features, I was first in the idea that it is the responsibility of quarkus-jackson to test them. But given, we already found an issue with @ JsonView, I wonder if it would make sense to test @ JsonInclude also ?

I think for camel-quarkus-jackson or camel-quarkus-jacksonxml,  if we cover all usecases in camel-jackson and camel-jacksonxml components, we should be good. Back to the @ JsonInclude, only NON_NULL is used in camel related components and this has been covered in the tests I added(the jacksonXmlIncludeNoNull test)

Cheers
Freeman

**Comment by ffang** [comment]
> Thanks, the test coverage is highly appreciated. Except for the two minor comments inline, could you please move the Jackson XML related stuff to separate test classes (JacksonXmlTest and JacksonXmlIT) and to a separate package (e.g. `org.apache.camel.quarkus.component.dataformats.jackson.xml`) in the test app? That would make it easier for the future readers to understand which parts are testing which extensions.

Thanks for the feedback! Will revise accordingly.

File: integration-tests/dataformats-json/pom.xml [hunk 3]
```diff
@@ -83,18 +87,29 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy-jsonb</artifactId>
         </dependency>
-
-        <!-- test dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit5</artifactId>
```
[the remark this answers is not in the document]
Comment (ffang) [reply integration-tests/dataformats-json/pom.xml:92]: will put this back to test scope

**Comment by ffang** [comment]
pushed a new commit which I believe addressed all feedback.
Thanks guys!

**Pull request merged**